Added GitLab integration example to docs (#1471)
earthly/earthly: docs/ci-integration/guides/gitlab-integration.md

# GitLab CI/CD integration

> Here is an example a GitLab CI/CD build, where we build the Earthly target `+build`.

Reviewer: NIT: avoid starting with "Here is"; maybe we can state it more concisely: 
```suggestion
This example uses [GitLab CI/CD](https://docs.gitlab.com/ee/ci/) to build the Earthly target `+build`.

```

```yml
# .gitlab-ci.yml

image: docker
services:
  - docker:dind

before_script:
    - apk update && apk add git
    - wget https://github.com/earthly/earthly/releases/download/v0.6.1/earthly-linux-amd64 -O /usr/local/bin/earthly
    - chmod +x /usr/local/bin/earthly
    - /usr/local/bin/earthly bootstrap
    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY

earthly:
  stage: build
  script:
    - earthly --ci --push -P +build
```

> For the full example on GitLab, see the [demo project here](https://gitlab.com/brandon176/earthly-demo/-/tree/main).

Reviewer: NIT: maybe we can avoid using the word `here` in the link, by wording it like:
```suggestion
A full example is available [on GitLab](https://gitlab.com/brandon176/earthly-demo/-/tree/main).
```

# GitLab CI/CD integration

This example uses [GitLab CI/CD](https://docs.gitlab.com/ee/ci/) to build the Earthly target `+build`.

```yml
# .gitlab-ci.yml

image: docker
services:
  - docker:dind

before_script:
    - apk update && apk add git

>     - wget https://github.com/earthly/earthly/releases/download/v0.6.1/earthly-linux-amd64 -O /usr/local/bin/earthly

Reviewer 2: Seems that this is one version behind - mind updating to `v0.6.2`? Can you also check `release/README.md` to see if the process to update versions in docs would cover this page too?

    - chmod +x /usr/local/bin/earthly
    - /usr/local/bin/earthly bootstrap
    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY

earthly:
  stage: build
  script:

>     - earthly --ci --push -P +build

Reviewer 2: Seems that the output is not colorized. Perhaps it's worth adding `FORCE_COLOR=1` in the env.

```

> A full example is available [on GitLab](https://gitlab.com/brandon176/earthly-demo/-/tree/main).

Reviewer 2: Should this be under an `earthly` GitLab user perhaps?
